src/fmu/dataio/providers/objectdata/_base.py

```diff
@@ -80,53 +79,44 @@ def __post_init__(self) -> None:
         named_stratigraphy = self._get_named_stratigraphy()
-
         self.name = named_stratigraphy.name
-        self._metadata["name"] = self.name
-        self._metadata["stratigraphic"] = named_stratigraphy.stratigraphic
-        self._metadata["offset"] = named_stratigraphy.offset
-        self._metadata["alias"] = named_stratigraphy.alias
-        self._metadata["top"] = named_stratigraphy.top
-        self._metadata["base"] = named_stratigraphy.base
-
-        self._metadata["content"] = (usecontent := content_model.content)
+
+        metadata: dict[str, Any] = {}
+        metadata["name"] = self.name
+        metadata["stratigraphic"] = named_stratigraphy.stratigraphic
+        metadata["offset"] = named_stratigraphy.offset
+        metadata["alias"] = named_stratigraphy.alias
+        metadata["top"] = named_stratigraphy.top
+        metadata["base"] = named_stratigraphy.base
+
+        metadata["content"] = (usecontent := content_model.content)
         if content_model.content_incl_specific:
-            self._metadata[usecontent] = getattr(
+            metadata[usecontent] = getattr(
                 content_model.content_incl_specific, usecontent, None
             )
 
-        self._metadata["tagname"] = self.dataio.tagname
-        self._metadata["format"] = self.fmt
-        self._metadata["layout"] = self.layout
-        self._metadata["unit"] = self.dataio.unit or ""
-        self._metadata["vertical_domain"] = list(self.dataio.vertical_domain.keys())[0]
-        self._metadata["depth_reference"] = list(self.dataio.vertical_domain.values())[
-            0
-        ]
-
-        self._metadata["spec"] = (
-            spec.model_dump(mode="json", exclude_none=True)
-            if (spec := self.get_spec())
-            else None
-        )
-        self._metadata["bbox"] = (
-            bbox.model_dump(mode="json", exclude_none=True)
-            if (bbox := self.get_bbox())
-            else None
+        metadata["tagname"] = self.dataio.tagname
+        metadata["format"] = self.fmt
+        metadata["layout"] = self.layout
+        metadata["unit"] = self.dataio.unit or ""
+        metadata["vertical_domain"] = list(self.dataio.vertical_domain.keys())[0]
+        metadata["depth_reference"] = list(self.dataio.vertical_domain.values())[0]
+
+        metadata["spec"] = self.get_spec()
+        metadata["bbox"] = self.get_bbox()
+        metadata["time"] = self._get_timedata()
+        metadata["table_index"] = self.table_index
+        metadata["undef_is_zero"] = self.dataio.undef_is_zero
+
+        metadata["is_prediction"] = self.dataio.is_prediction
+        metadata["is_observation"] = self.dataio.is_observation
+        metadata["description"] = generate_description(self.dataio.description)
+
+        self._metadata = (
+            UnsetAnyContent.model_validate(metadata)
+            if metadata["content"] == "unset"
+            else AnyContent.model_validate(metadata)
         )
```

Comment:
Really eager to get away from the `UnsetAnyContent` hack here and populate the `AnyContent` more directly if possible.
 The metadata created without content is not possible to upload to sumo, so not sure if I see any need for this functionality anymore. But I don't have the full overview of why it was added 🙂 I guess before when using `ExportData` to re-export preprocessed data it was annoying to have to give in a content that was not going to be used, but now with the new `ExportPreprocessedData` class it will not be an issue anymore

Reply:
Agreed, I packed into a dict initially for the reason but really do not like it. I created #688